main.tf

```diff
@@ -21,6 +21,13 @@ resource "aws_elasticache_replication_group" "redis" {
   maintenance_window            = var.redis_maintenance_window
   snapshot_window               = var.redis_snapshot_window
   snapshot_retention_limit      = var.redis_snapshot_retention_limit
+  availability_zones            = var.availability_zones
+
+  cluster_mode {
```

Comment:
Before adding a dynamic block like this without a conditional because it will impact all existing codebase using this module. Please add a conditional in the block so that the code is not applied if a specific variable is not present. To test it works correctly try to plan a terraform state that uses this repo and see that there is "no chancge".